requests_negotiate_sspi/requests_negotiate_sspi.py

```diff
@@ -53,7 +54,7 @@ def _retry_using_http_Negotiate_auth(self, response, scheme, args):
 
         if self._host is None:
             targeturl = urlparse(response.request.url)
-            self._host = targeturl.hostname
+            self._host = socket.gethostbyname_ex(targeturl.hostname)[0] # canonicalize hostname
```

Comment:
According to [the documentation](https://docs.python.org/2/library/socket.html#socket.gethostbyname_ex), gethostbyname_ex does not support IPv6. Can you change this to something like:
```python
socket.getaddrinfo(targeturl.hostname, None)[3]
```
Not sure if this might also merit a try/except block that falls back to the old behavior if the hostname cannot be canonicalized for any reason?